unlock-app/src/middlewares/providerMiddleware.ts

```diff
@@ -22,13 +33,41 @@ function initializeProvider(provider: { enable?: () => any }, dispatch: any) {
       .catch(() =>
         dispatch(setError(Application.Fatal(FATAL_NOT_ENABLED_IN_PROVIDER)))
       )
+  } else if (provider.isUnlock) {
+    // This initialization is handled on receipt of GOT_ENCRYPTED_PRIVATE_KEY_PAYLOAD
+    return
   } else {
     // Default case, provider doesn't have an enable method, so it must already be ready
     dispatch(providerReady())
   }
 }
 
-const providerMiddleware = (config: any) => {
+export async function initializeUnlockProvider(
+  action: Action,
+  unlockProvider: any,
+  dispatch: any
+) {
+  const { key, emailAddress, password } = action
+  try {
+    await unlockProvider.connect({ key, password })
+
+    const address = unlockProvider.wallet.address
+    dispatch(setAccount({ address }))
+    dispatch(setEncryptedPrivateKey(key, emailAddress))
+    dispatch(providerReady())
+  } catch (_) {
+    // TODO: password isn't the only thing that can go wrong here...
```

Comment:
can we console.error the error? it makes debugging ever so much more possible

Reply:
Do our lint rules not complain about that? There are actually a bunch of places where we could log that out if it's allowed.

Reply:
oh if you do:

```js
// eslint-disable-next-line no-console
console.error(e)
```

Then lint doesn't complain :)

Reply:
I think this is a good use case for breaking the lint rules, in other words